### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by izaiasneto4**
# Resume

We created the methods as described in the project requirements, added some basic tests, and *came to the conclusion that we met all of the requirements*.

# Proposal

We want to have a review of the code to know if we need to make any changes. If no changes needed we want the project to be ACCEPTED so we can move to the next project with a happy smile :)

# Comments

Working on ruby is interesting because of new concepts, but is a good thing that ruby have a lot of good utilities that helps a lot. It is an interesting language to work with.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    end
+    break if sorted
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  if verify_input(arr)
+    if block_given?
+      loop do
```
Comment (uche-inyama) [inline main.rb]: check your block, the error states: expected Exception but nothing was raised

        bubble_sort_by raises an error if block is not given
        Failure/Error: expect { bubble_sort_by(unsorted) }.to raise_error expected Exception but nothing was raised

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Hi team!

Your methods passed the test cases, expect one:

- [ ] rspec ./spec/bubble_sort_spec.rb:19 # bubble_sort_by raises an error if block is not given. Check the exception you raised.

After rectifying the issue, submit another form for a code review.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

File: main.rb (hunk 1 above)
Comment (Israel-Laguan) [reply main.rb]: Oh, ok, the problem is that we can't find the tests that you are referring. Can you tell us where are such tests please? Also there was nothing about exceptions on the requirements, but we can add that exception without problem. Thanks in advance!

**Review by Ispirett — APPROVED** [review]
## Status Approved

Great job @izaiasneto4  @Israel-Laguan  you can merge this project now.  :+1:  :100:

**Pull request merged**